plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1531,25 +1531,39 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
     public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column)
     {
         IcebergColumnHandle handle = (IcebergColumnHandle) column;
+        dropColumn(session, tableHandle, handle.getName());
+    }
+
+    @Override
+    public void dropField(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column, List<String> fieldName)
+    {
+        IcebergColumnHandle handle = (IcebergColumnHandle) column;
+        String name = String.join(".", ImmutableList.<String>builder().add(handle.getName()).addAll(fieldName).build());
```

Comment:
Does this have ambiguity problem?

Let's have tests

```sql
CREATE TABLE ...(a row(b int, c int), "a.b" int);

ALTER ... DROP COLUMN a.b; -- this should drop the field "b" in the "a" column of type row
```


```sql
CREATE TABLE ...(a row(b int, c int), "a.b" int);

ALTER ... DROP COLUMN "a.b"; -- this should drop the column "a.b"
```

Reply:
We can't create such tables in both Trino and Spark. It's ensured by `BaseIcebergConnectorTest#testAmbiguousColumnsWithDots`.

Reply:
Add a code comment please.